crypto/store/store_lib.c

```diff
@@ -1095,3 +1095,62 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bp, const char *scheme,
 
     return ctx;
 }
+
+const char *OSSL_file_stat(const char *uri, struct stat *st)
+{
+    const char *path = uri, *q;
+    struct stat local_st;
+
+    if (st == NULL)
+        st = &local_st;
+
+    /*
+     * First, unless the URI starts with "file://",
+     * try and see if the full URI can be taken as a local file path name.
+     */
+    if (!HAS_CASE_PREFIX(uri, "file://")) {
+        if (stat(path, st) == 0)
+            return uri;
+        ERR_raise_data(ERR_LIB_SYS, errno, "calling stat(%s)", path);
+    }
+
+    /* Do a second attempt only if the URI appears to start with the "file" scheme. */
+    if (!CHECK_AND_SKIP_CASE_PREFIX(path, "file:"))
+        return NULL;
+
```

Comment:
Of course, I understand that without the "file://" check above, this condition turns out to be bad too.  So I'd drop this too.

Reply:
In total, I just followed the existing semantics.

If we dropped both `!HAS_CASE_PREFIX(uri, "file://")` and this condition here,
we would unconditionally try at first interpreting the whole URI as a local file path name also in case there is an authority given (be it "localhost" or anything else), which would not make sense to me.
Do you really propose such a change?

Reply:
That was fixed like this:

``` C
        if (CHECK_AND_SKIP_CASE_PREFIX(q, "//")) {
            path_data_n--;           /* Invalidate using the full URI */
```

You could do something similar with by just assigning NULL to `uri`, and only do stat() on `uri` when it's non-NULL.

Reply:
That would again make the code needlessly complicated, thus hard to understand and maintain.

Reply:
That complicated original code is the very reason for the reviewing trouble right now.

Reply:
... and never mind this discussion, got it now